packages/rrweb-snapshot/src/snapshot.ts

```diff
@@ -579,16 +710,48 @@ function serializeTextNode(
   if (isScript) {
     textContent = 'SCRIPT_PLACEHOLDER';
   }
-  if (
-    !isStyle &&
-    !isScript &&
-    textContent &&
-    needMaskingText(n, maskTextClass, maskTextSelector)
-  ) {
+  const forceMask = needMaskingText(
+    n,
+    maskTextClass,
+    maskTextSelector,
+    unmaskTextClass,
+    unmaskTextSelector,
+    maskAllText,
+  );
+
+  if (!isStyle && !isScript && textContent && forceMask) {
```

Comment:
should we also check `&& !isTextarea` here? 🤔 could it be double-masked otherwise?